packages/arti/build.sh

```diff
@@ -0,0 +1,36 @@
+TERMUX_PKG_HOMEPAGE=https://tpo.pages.torproject.net/core/arti/
+TERMUX_PKG_DESCRIPTION="Arti is a work-in-progress project to write a full-featured Tor implementation in Rust."
+TERMUX_PKG_LICENSE="Apache-2.0, MIT"
+TERMUX_PKG_MAINTAINER="Yaksh Bariya <thunder-coding@termux.dev>"
+TERMUX_PKG_VERSION="1.2.8"
+TERMUX_PKG_SRCURL="https://gitlab.torproject.org/tpo/core/arti/-/archive/arti-v${TERMUX_PKG_VERSION}/arti-arti-v${TERMUX_PKG_VERSION}.tar.gz"
+TERMUX_PKG_SHA256=85dd949c4ac29d9b13f599c65de17dcd2b60ba6963dab27be7b5530a0bfe7675
+TERMUX_PKG_DEPENDS="liblzma, libsqlite, openssl"
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_pre_configure() {
+	termux_setup_rust
+}
+
+termux_step_make() {
+	cargo build --jobs $TERMUX_PKG_MAKE_PROCESSES --target $CARGO_TARGET_NAME --release --features full
+}
+
+termux_step_make_install() {
+	install -Dm755 -t $TERMUX_PREFIX/bin target/${CARGO_TARGET_NAME}/release/arti
+	install -Dm640 -t $TERMUX_PREFIX/etc/arti.d/arti.toml crates/arti/src/arti-example-config.toml
+}
+
+termux_step_create_debscripts() {
+	cat <<- EOF > ./postinst
+	echo
+	echo "-------------------------WARNING-------------------------"
+	echo
+	echo "Arti is an EXPERIMENTAL implementation of existing CTor."
+	echo
+	echo "It is not recommended to use Arti unless you know what you are doing. It may not be up to mark with existing ctor in terms of stability, performance and MOST IMPORTANTLY SECURITY. Use Arti only if you want to help find bugs in the software"
```

Comment:
Maybe something like:
```suggestion
	echo "It is not recommended to use Arti unless you have previous experience with Tor."
	echo "It may not be up to parity with existing ctor in terms of stability, performance and MOST IMPORTANTLY SECURITY."
	echo "Treat Arti moreso as an experimental package provided to find bugs in the project."
```

Reply:
That's not nearly strong enough. The warning here is that this implementation is WIP and may wildly fail to provide the guarantees you're expecting of tor.

@thunder-coding I can offer some tweaks to the language, but first: tor is probably the closest thing we ship to safety-critical software. People do things that risk physical harm using tor because they believe it's trustworthy. The devs of this implementation are extremely clear that you should not be using it yet unless you are actively developing tor. Presumably, you felt the loudly repeat that in a postinst because you are aware that packaging something implies a high degree of out-of-the-box readiness.

If the only people who should be using this are developers, why do you want to package this, and is it a good idea to do so? My immediate read on this situation is "heck no".

Reply:
Thanks, converting this PR to draft. Will keep on updating versions of arti as they come. Will start shipping this when Tor devs announce about declaring arti to be stable enough!

Reply:
https://github.com/termux/termux-packages/pull/21993#issuecomment-2973545384